tools/plugins/com.liferay.ide.upgrade.core/src/com/liferay/blade/api/XMLFile.java

```diff
@@ -23,4 +23,6 @@
 
 	public Collection<SearchResult> findElement(String elementName, String elementValue);
 
+	public SearchResult getDocumentTypeDeclaration(String version, String filter);
```

Comment:
```suggestion
	public SearchResult findDocumentTypeDeclaration(String name, String idPattern);
```
Lets use this API instead.  You pass the name of the doctype element (must match `IDOMDocument.getDocumentType().getName()` and then the 2nd parameter is a regex that can be used to match the PUBLIC ID for DTDs.